VirtoCommerce.Domain/Search/IIndexingManager.cs

```diff
@@ -23,6 +22,40 @@ public interface IIndexingManager
         /// <param name="progressCallback"></param>
         /// <param name="cancellationToken"></param>
         /// <returns></returns>
-        Task IndexAsync(IndexingOptions options, Action<IndexingProgress> progressCallback, CancellationToken cancellationToken);
+        Task IndexAsync(IndexingOptions options, Action<IndexingProgress> progressCallback, Platform.Core.Common.ICancellationToken cancellationToken);
+
+        /// <summary>
+        /// Indexes a batch of documents through the queue.
+        /// </summary>
+        /// <param name="documentType">Document type to index.</param>
+        /// <param name="documentIds">Ids of documents to index.</param>
+        /// <param name="priority">Priority for the indexation work.</param>
+        void QueueDocumentIndexation(string documentType, string[] documentIds, IndexingPriority priority = IndexingPriority.Default);
```

Comment:
Could we move indexing scale-out functionality from IndexingManager, to Controller action or IndexJob? In my opinion, this violates single responsibility principle for the Index manager?

Reply:
You're right, have removed the methods. Job queue is responsibility of IndexingJobs class.